test/specs/originalEvtSpec.js

```diff
@@ -0,0 +1,99 @@
+describe("Original Event trasmittance test", function () {
+    var SLIDER_ID = "testSlider1";
+    var slider;
+    var options;
+    var origEvtInSlidestart, origEvtInSlide, origEvtInSlidestop;
+
+	function slider_callback(evt_type) {
+		return function (val, evt) {
+			if (evt && evt.type && evt.type === evt_type) {
+				// check to which event this callback is bound:
+				switch (evt.type) {
+					case "mousedown":
+						origEvtInSlidestart = 1;
+						break;
+					case "mousemove":
+						origEvtInSlide = 1;
+						break;
+					case "mouseup":
+						origEvtInSlidestop = 1;
+						break;
+					default:
+						// do nothing.
+				}
+                    origEvtInSlidestart = 1;
+            }
+		};
+	}
+    
+    describe('When a slider w/o jQuery is dragged', function () {
+        beforeEach(function () {
+            options = {
+                id: "myslider",
+                min: 0,
+                max: 100,
+                value: 0
+            };
+            slider = new Slider(document.getElementById(SLIDER_ID), options);
+            origEvtInSlidestart = 0;
+            origEvtInSlide = 0;
+            origEvtInSlidestop = 0;
+
+            slider.on("slideStart", slider_callback("mousedown"));
+            slider.on("slide", slider_callback("mousemove"));
+            slider.on("slideStop", slider_callback("mouseup"));
+
+            var sliderLeft = $("#myslider")[0].offsetLeft;
+            var offsetY = $("#myslider")[0].offsetTop;
+            var offsetX = sliderLeft + $("#myslider")[0].clientWidth;
+            var mouseDownEvent = getMouseEvent('mousedown', offsetX, offsetY);
+            var mouseMoveEvent = getMouseEvent('mousemove', offsetX - 20, offsetY);
+            var mouseUpEvent = getMouseEvent('mouseup', offsetX - 20, offsetY);
+            slider.mousedown(mouseDownEvent);
+            slider.mousemove(mouseMoveEvent);
+            slider.mouseup(mouseUpEvent);
+        });
+
+        it("slideStart event should contain mousedown evt", function () {
+            expect(origEvtInSlidestart).toBe(1);
+        });
+        it("slide event should contain mousemove evt", function () {
+            expect(origEvtInSlide).toBe(1);
+        });
+        it("slideStop event should contain mouseup evt", function () {
+            expect(origEvtInSlidestop).toBe(1);
+        });
+    });
+
+    afterEach(function () {
+        slider.destroy();
+    });
+
+    function getMouseEvent(type, offsetXToClick, offsetYToClick) {
+        var args = {
+            clientX: offsetXToClick,
+            clientY: offsetYToClick,
+            ctrlKey: false,
+            shiftKey: false,
+            altKey: false
+        };
+        var event = new MouseEvent(type, args);
+        return event;
+    }
+
+    // MouseEvent constructor  polyfill
+    function polyfill(window) {
+        var MouseEvent3 = function (eventType, params) {
```

Comment:
sorry for the late reply @drhouse82 but this all looks awesome!

One last request...could you move this polyfill to a separate file, add it to the package.json, and then include it in the jasmine test page via the grunt file?

See the `phantom_bind_polyfill.js` as an example:

* https://github.com/seiyria/bootstrap-slider/blob/master/test/phantom_bind_polyfill.js
* `package.json reference:` https://github.com/seiyria/bootstrap-slider/blob/286f98389971bc521364778205e2d704a4cefb3d/package.json#L70
* `Gruntfile reference:` https://github.com/seiyria/bootstrap-slider/blob/458d7ab6c4e776a9c0bfeff7e526591ac4191b82/Gruntfile.js#L125